src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Get.cs

```diff
@@ -73,6 +73,10 @@ public static partial class HttpClientJsonExtensions
             using (HttpResponseMessage response = await taskResponse.ConfigureAwait(false))
             {
                 response.EnsureSuccessStatusCode();
+                if (response.StatusCode == System.Net.HttpStatusCode.NoContent)
+                {
+                    return default;
+                }
```

Comment:
Why would we special-case it?  If the server sends back 204 in response to something the caller is expecting to get content from, why pretend something else happened?

Reply:
Ok, gents, I would like to explain the idea by an example

Imagine that we have a micro-service implement by dotnet or any other fancy tech, it provides an API method on route path `/dog/{id}` and returns:
1. `HTTP 200` and a valid JSON string of a dog
2. Or `HTTP 204` and no content.

Now let us consume the API by using the extension method `GetFromJsonAsync`, it works well for HTTP 200 responses, but it throws an exception for HTTP 204. 

HTTP 204 is a reasonable response when the dog id does not exist, it makes more sense if the `GetFromJsonAsync` returns `null`.

And I think this is a common use case.

Thank you for reviewing!

Reply:
The better pattern here is likely for you to get a response message, check the status code, and then decode the content based on status code. Beyond letting you handle 204, this would allow you to pull out a different model if you get a validation error, etc.